### PR TITLE
[Lang] Check version when importing taichi instead of when using ti

### PR DESCRIPTION
**Pull request by Leonz5288**
Related issue = #

### CODE REVIEW

File: python/taichi/lang/__init__.py [hunk 1]
```diff
@@ -456,6 +523,26 @@ def init(arch=None,
...
+            last_time = f.readlines()[0].rstrip()
+        if cur_date.strftime('%Y-%m-%d') > last_time:
+            check_version()
+            with open(timestamp_path, 'w') as f:
+                f.write((cur_date +
+                         datetime.timedelta(days=14)).strftime('%Y-%m-%d'))
+                f.truncate()
+    else:
+        check_version()
+        with open(timestamp_path, 'w') as f:
```
Comment (rexwangcc) [inline python/taichi/lang/__init__.py:542]:
```suggestion
        with open(timestamp_path, 'wb') as f:
```
maybe use binary mode? (if so, same for all R/W...)

**Review by rexwangcc — COMMENTED** [review]
Looks great and thanks! I left one comment, and guess @ailzhang could provide insights about the new deps (requests).

File: python/taichi/lang/__init__.py (hunk 1 above)
Comment (Leonz5288) [reply python/taichi/lang/__init__.py:542]: Is binary mode better?

**Review by lin-hitonami — APPROVED** [review]
LGTM！

**Pull request merged**